src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,34 @@
 package core.basesyntax;
 
+import core.utils.DateFormatter;
+import java.time.LocalDate;
+
 public class SalaryInfo {
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        StringBuilder builder = new StringBuilder("Report for period " + dateFrom + " - " + dateTo);
+
+        for (String name : names) {
+            int totalSalary = 0;
+            for (String dates : data) {
+                if (dates.contains(name)) {
```

Comment:
Using 'contains' method to check if the string contains the name is not reliable.
Will your program work correct if you have John Smith and John Smith Jr. among employees?